be/src/exec/pipeline/scan/connector_scan_operator.cpp

```diff
@@ -179,7 +179,11 @@ ConnectorChunkSource::~ConnectorChunkSource() {
 Status ConnectorChunkSource::prepare(RuntimeState* state) {
     RETURN_IF_ERROR(ChunkSource::prepare(state));
     _runtime_state = state;
-    _ck_acc.set_max_size(state->chunk_size());
+    if (_limit != -1 && _limit < state->chunk_size()) {
```

Comment:
the predicates must be empty. To eliminate statements like `select * from gender = "male" limit 10` otherwise it will choose a small chunk size for it.

Reply:
added the logic to check predicate